### PR TITLE
task changes

### PR DESCRIPTION
**Pull request by AshwiniChalla**
Testing : manual

### CODE REVIEW

**Comment by AshwiniChalla** [comment]
Description : changes to add support for publish code coveragetask for xplat

File: externals.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-    "vsts-task-lib":"0.5.10",
+    "vsts-task-lib":"0.6.1",
```
Comment (acesiddhu) [inline externals.json]: can u get a sign offs on this from build team?

**Comment by AshwiniChalla** [comment]
Added Unit tests

File: Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+var summaryFileLocation = tl.getInput('summaryFileLocation', true);
+var reportDirectory = tl.getInput('reportDirectory');
+var additionalCodeCoverageFiles = tl.getInput('additionalCodeCoverageFiles');
+
+tl.debug('codeCoverageTool: ' + codeCoverageTool);
+tl.debug('summaryFileLocation: ' + summaryFileLocation);
+tl.debug('reportDirectory: ' + reportDirectory);
+tl.debug('additionalCodeCoverageFiles: ' + additionalCodeCoverageFiles);
+
+if(additionalCodeCoverageFiles && (additionalCodeCoverageFiles.indexOf('*') >= 0 || additionalCodeCoverageFiles.indexOf('?') >= 0)) {
```
Comment (nigurr) [inline Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: what's ? represents

File: Tests/L0/PublishCodeCoverageResults/_suite.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+		
+		var tr = new trm.TaskRunner('PublishCodeCoverageResults');
+        
+		tr.setInput('codeCoverageTool', 'JaCoCo');
+		tr.setInput('summaryFileLocation', '/user/admin/summary.xml');
+		tr.setInput('additionalCodeCoverageFiles', "/some/*pattern");
+        
+		tr.run()
+		.then(() => {
+			assert(tr.stderr.length == 0, 'should not have written to stderr. error: ' + tr.stderr);
```
Comment (nigurr) [inline Tests/L0/PublishCodeCoverageResults/_suite.ts]: indention

**Comment by nigurr** [comment]
Update tool tip information

File: Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts (hunk 2 above)
Comment (AshwiniChalla) [inline Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: single character I guess

**Comment by AshwiniChalla** [comment]
there is noting specific to windows in tool tips

File: Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts (hunk 2 above)
Comment (acesiddhu) [inline Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: can u explain this? what are we checking for here?

File: Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  var buildFolder = tl.getVariable('agent.buildDirectory');
+  var allFiles = tl.find(buildFolder);
+  var codeCoverageFiles = tl.match(allFiles, additionalCodeCoverageFiles, { matchBase: true });
+}
+else if(additionalCodeCoverageFiles) {
+  var  codeCoverageFiles = [additionalCodeCoverageFiles];
+}
+
+var tp = new tl.CodeCoveragePublisher();
```
Comment (acesiddhu) [inline Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: rename tp to something meaningful

File: externals.json [hunk 5]
```diff
@@ -1,4 +1,4 @@
 {
-    "vsts-task-lib":"0.5.10",
+    "vsts-task-lib":"0.6.0",
```
Comment (bryanmacfarlane) [inline externals.json]: I just published this 0.6.0 so we're good on this specific change

File: Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts (hunk 2 above)
Comment (AshwiniChalla) [inline Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: ? and * mark are the regex characters. If they are present in the additionalCoverageFiles we try to fliter them. If not we directly add it to array assuming it is a file name without any regex chars

**Comment by AshwiniChalla** [comment]
Documentaton changes will be done once the vso-agent is released with related changes

**Pull request merged**